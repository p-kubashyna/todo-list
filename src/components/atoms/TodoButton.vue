<script setup>
const props = defineProps({
  label: {
    type: String,
    required: true,
    default: "",
  },
  isDisabled: {
    type: Boolean,
    required: false,
    default: false,
  },
  additionalClass: {
    type: String,
    required: false,
    default: "",
  },
});
</script>

<template>
  <button
    class="default-button"
    :disabled="props.isDisabled"
    :class="{
      [props.additionalClass]: props.additionalClass !== '',
      'active-button': !props.isDisabled,
    }"
  >
    {{ props.label }}
  </button>
</template>

<style>
.default-button {
  background: white;
  margin-left: 5px;
  font-family: monospace;
  border-style: none;
}

.active-button:hover {
  text-decoration: underline;
}
</style>
