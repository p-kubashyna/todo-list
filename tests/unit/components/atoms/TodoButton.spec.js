import { describe, expect, it } from "@jest/globals";
import { shallowMount } from "@vue/test-utils";
import TodoButton from "@/components/atoms/TodoButton.vue";

describe("Given the TodoButton component", () => {
  it.each([
    [
      {
        label: "button 1",
        isDisabled: false,
        additionalClass: "additional-class",
      },
    ],
    [
      {
        label: "button 2",
        isDisabled: true,
        additionalClass: "additional-class",
      },
    ],
    [{ label: "button 3", isDisabled: false, additionalClass: "" }],
    [{ label: "button 4" }],
  ])("should render", (propsData) => {
    const wrapper = shallowMount(TodoButton, { props: propsData });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("should disable button when disabled prop is passed", async () => {
    const wrapper = shallowMount(TodoButton, {
      props: { label: "button", isDisabled: true },
    });
    const button = wrapper.find("button");
    expect(button.element.disabled).toBe(true);
  });
});
