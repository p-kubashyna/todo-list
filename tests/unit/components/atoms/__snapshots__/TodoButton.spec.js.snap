// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the TodoButton component should render 1`] = `
<button
  class="default-button additional-class active-button"
>
  button 1
</button>
`;

exports[`Given the TodoButton component should render 2`] = `
<button
  class="default-button additional-class"
  disabled=""
>
  button 2
</button>
`;

exports[`Given the TodoButton component should render 3`] = `
<button
  class="default-button active-button"
>
  button 3
</button>
`;

exports[`Given the TodoButton component should render 4`] = `
<button
  class="default-button active-button"
>
  button 4
</button>
`;
